racket/src/racket/src/jitinline.c

```diff
@@ -1283,6 +1284,12 @@ int scheme_generate_inlined_unary(mz_jit_state *jitter, Scheme_App2_Rec *app, in
   } else if (IS_NAMED_PRIM(rator, "path?")) {
     generate_inlined_type_test(jitter, app, SCHEME_PLATFORM_PATH_KIND, SCHEME_PLATFORM_PATH_KIND, 0, for_branch, branch_short, dest);
     return 1;
+  } else if (IS_NAMED_PRIM(rator, "hash?")) {
+    generate_inlined_type_test(jitter, app, scheme_hash_table_type, scheme_hash_tree_indirection_type, 1, for_branch, branch_short, dest);
+    return 1;
+  } else if (IS_NAMED_PRIM(rator, "syntax?")) {
+    generate_inlined_type_test(jitter, app, scheme_primitive_syntax_type, scheme_primitive_syntax_type, 0, for_branch, branch_short, dest);
```

Comment:
This one should be `scheme_stx_type`.
The `syntax?` primitive function has to be marked as inlineable for these to take effect, so that's probably why you're not seeing failures.